core/src/broadcast_stage/broadcast_duplicates_run.rs

```diff
@@ -269,44 +273,58 @@ impl BroadcastRun for BroadcastDuplicatesRun {
                 .map(|(pubkey, _)| *pubkey)
                 .collect()
         };
+
         // Broadcast data
         let cluster_nodes =
             self.cluster_nodes_cache
                 .get(slot, &root_bank, &working_bank, cluster_info);
         let socket_addr_space = cluster_info.socket_addr_space();
         let packets: Vec<_> = shreds
             .iter()
-            .filter_map(|shred| {
+            .flat_map(|shred| {
```

Comment:
Probably better to keep `filter_map` here so that we have the `?` operator inside, and just add `.flatten()` right before `.collect()` at the end.